types/client/index.d.ts

```diff
@@ -1,10 +1,64 @@
 // Minimum TypeScript Version: 3.8
+import { IncomingMessage } from 'http';
+
 interface Session {
-  foo: string;
+  user: {
+    name: string;
+    email: string;
+    image: string;
+  };
+  accessToken: string;
+  expires: string;
+}
+
+interface GetProvidersResponse {
+  [provider: string]: Provider;
+}
+
+interface Provider {
+  id: string;
+  name: string;
+  type: string;
+  signinUrl: string;
+  callbackUrl: string;
+}
+
+interface GenericObject {
+  [key: string]: any;
 }
 
 type UseSessionResult = [Session, boolean];
 
 declare function useSession(): UseSessionResult;
+declare function getSession(context: NextContext): Session | null;
```

Comment:
These functions return promises

Reply:
Good catch 👍🏻 , I actually need to export as well the declarations for the aliases as well ( `session`, `crsfToken` and `providers` ) 💆🏻‍♂️